src/Oracle/Oracle.cs

```diff
@@ -0,0 +1,377 @@
+
+            Log($"Builded response tx:{responseTx.Hash} requestTx:{request.OriginalTxid} requestId: {requestId}");
+
+            ECPoint[] oraclePublicKeys = NativeContract.Oracle.GetOracleNodes(snapshot);
+            foreach (var account in Wallet.GetAccounts())
+            {
+                var oraclePub = account.GetKey().PublicKey;
+                if (!account.HasKey || account.Lock || !oraclePublicKeys.Contains(oraclePub)) continue;
+
+                var txSign = responseTx.Sign(account.GetKey());
+                AddResponseTxSign(snapshot, requestId, txSign, oraclePub, responseTx);
+                SendResponseSignature(requestId, txSign, account.GetKey());
+
+                Log($"Send oracle sign data: Oracle node: {oraclePub} RequestTx: {request.OriginalTxid} Sign: {txSign.ToHexString()}");
+            }
+        }
+
+        private Transaction CreateResponseTx(StoreView snapshot, OracleResponse response)
+        {
+            var oracleNodes = NativeContract.Oracle.GetOracleNodes(snapshot);
+            var m = oracleNodes.Length - (oracleNodes.Length - 1) / 3;
+            var n = oracleNodes.Length;
+            var oracleSignContract = Contract.CreateMultiSigContract(m, oracleNodes);
+
+            var tx = new Transaction()
+            {
+                Version = 0,
+                ValidUntilBlock = snapshot.Height + Transaction.MaxValidUntilBlockIncrement,
+                Attributes = new TransactionAttribute[] {
+                    response
+                },
+                Signers = new Signer[]
+                {
+                    new Signer()
+                    {
+                        Account = NativeContract.Oracle.Hash,
+                        AllowedContracts = new UInt160[] { },
+                        Scopes = WitnessScope.None
+                    },
+                    new Signer()
+                    {
+                        Account = oracleSignContract.ScriptHash,
+                        AllowedContracts = new UInt160[] { NativeContract.Oracle.Hash },
+                        Scopes = WitnessScope.None
+                    },
+                },
+                Witnesses = new Witness[2],
+                Script = OracleResponse.FixedScript,
+                NetworkFee = 0,
+                Nonce = 0,
+                SystemFee = 0
+            };
+            Dictionary<UInt160, Witness> witnessDict = new Dictionary<UInt160, Witness>();
+            witnessDict[oracleSignContract.ScriptHash] = new Witness
+            {
+                InvocationScript = Array.Empty<byte>(),
+                VerificationScript = oracleSignContract.Script,
+            };
+            witnessDict[NativeContract.Oracle.Hash] = new Witness
+            {
+                InvocationScript = new ScriptBuilder().EmitPush(0).Emit(OpCode.PACK).EmitPush("verify").ToArray(),
+                VerificationScript = NativeContract.Oracle.Script,
+            };
```

Comment:
If we have empty verification script, it will automatically call contracts `verify` method based on `Signers` array.
https://github.com/neo-project/neo/blob/914b435426af745a627b1080494c88ad2a5173e5/src/neo/SmartContract/Helper.cs#L160

Also what is the purpose of calling `verify` in `InvocationScript`?

Reply:
Done